dbt/adapters/greenplum/impl.py

```diff
@@ -1,135 +1,30 @@
-from datetime import datetime
-from dataclasses import dataclass
 from typing import Optional, Set, List, Any
 from dbt.adapters.base.meta import available
-from dbt.adapters.base.impl import AdapterConfig
-from dbt.adapters.sql import SQLAdapter
 from dbt.adapters.greenplum import GreenplumConnectionManager
 from dbt.adapters.greenplum.relation import GreenplumColumn
 from dbt.adapters.greenplum.relation import GreenplumRelation
-from dbt.dataclass_schema import dbtClassMixin, ValidationError
-import dbt.exceptions
-import dbt.utils
-
+from dbt.adapters.postgres.impl import PostgresIndexConfig, PostgresConfig, PostgresAdapter
+from dbt.adapters.postgres import impl
 
 # note that this isn't an adapter macro, so just a single underscore
-GET_RELATIONS_MACRO_NAME = "greenplum_get_relations"
-
+impl.GET_RELATIONS_MACRO_NAME = "greenplum_get_relations"
 
-@dataclass
-class GreenplumIndexConfig(dbtClassMixin):
-    columns: List[str]
-    unique: bool = False
-    type: Optional[str] = None
 
-    def render(self, relation):
-        # We append the current timestamp to the index name because otherwise
-        # the index will only be created on every other run. See
-        # https://github.com/dbt-labs/dbt-core/issues/1945#issuecomment-576714925
-        # for an explanation.
-        now = datetime.utcnow().isoformat()
-        inputs = self.columns + [relation.render(), str(self.unique), str(self.type), now]
-        string = "_".join(inputs)
-        return dbt.utils.md5(string)
+class GreenplumIndexConfig(PostgresIndexConfig):
+    pass
 
-    @classmethod
-    def parse(cls, raw_index) -> Optional["GreenplumIndexConfig"]:
-        if raw_index is None:
-            return None
-        try:
-            cls.validate(raw_index)
-            return cls.from_dict(raw_index)
-        except ValidationError as exc:
-            msg = dbt.exceptions.validator_error_message(exc)
-            dbt.exceptions.raise_compiler_error(f"Could not parse index config: {msg}")
-        except TypeError:
-            dbt.exceptions.raise_compiler_error(
-                f"Invalid index config:\n"
-                f"  Got: {raw_index}\n"
-                f'  Expected a dictionary with at minimum a "columns" key'
-            )
 
+class GreenplumConfig(PostgresConfig):
+    pass
 
-@dataclass
-class GreenplumConfig(AdapterConfig):
-    unlogged: Optional[bool] = None
-    indexes: Optional[List[GreenplumIndexConfig]] = None
 
-
-class GreenplumAdapter(SQLAdapter):
+class GreenplumAdapter(PostgresAdapter):
     Relation = GreenplumRelation
```

Comment:
given that we haven't done anything new when defining `GreenplumRelation` this line can also be deleted, and `GreenplumAdapter.Relation` will be initialized with `PostrgesRelation`
```suggestion
```

Reply:
Done